exercises/practice/mazy-mice/src/test/java/MazeGeneratorTest.java

```diff
@@ -0,0 +1,259 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Ignore;
+
+import java.util.Arrays;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+
+public class MazeGeneratorTest {
+    private static final char EMPTY_CELL = ' ';
+    private static final Set<Character> ALLOWED_SYMBOLS = Set.of(
+            EMPTY_CELL, // space
+            '┌', // box drawings light down and right
+            '─', // box drawings light horizontal
+            '┬', // box drawings light down and horizontal
+            '┐', // box drawings light down and left
+            '│', // box drawings light vertical
+            '└', // box drawings light up and right
+            '┴', // box drawings light up and horizontal
+            '┘', // box drawings light up and left
+            '├', // box drawings light vertical and right
+            '┤', // box drawings light vertical and left
+            '┼', // box drawings light vertical and horizontal
+            '⇨'  // rightwards white arrow
+    );
+    private static final char VISITED_CELL = '.';
+    private static final int RECTANGLE_ROWS = 6;
+    private static final int RECTANGLE_COLUMNS = 18;
+    private static final int SEED_ONE = 42;
+    private static final int SEED_TWO = 43;
+    private MazeGenerator sut;
+
+    @Before
+    public void setup() {
+        sut = new MazeGenerator();
+    }
+
+    @Test
+    public void mazeIsNotNull() {
+        var maze = sut.generatePerfectMaze(RECTANGLE_ROWS, RECTANGLE_COLUMNS);
+
+        assertThat(maze)
+                .as("The maze is not null.")
+                .isNotNull();
+
+        Arrays.stream(maze)
+                .map(String::new)
+                .forEach(System.out::println);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void theDimensionsOfTheMazeAreCorrect() {
+        var maze = sut.generatePerfectMaze(RECTANGLE_ROWS, RECTANGLE_COLUMNS);
+        var expectedWidth = RECTANGLE_COLUMNS * 2 + 1;
+        var expectedHeight = RECTANGLE_ROWS * 2 + 1;
+
+        assertThat(maze)
+                .as("The dimensions of the maze are correct.")
+                .hasSize(expectedHeight);
+
+        assertThat(maze[0])
+                .as("The dimensions of the maze are correct.")
+                .hasSize(expectedWidth);
```

Comment:
Since `maze` is a jagged 2D array, wouldn't it be better to assert that all rows have the correct number of columns?

Reply:
> Since `maze` is a jagged 2D array, wouldn't it be better to assert that all rows have the correct number of columns?

Can I update AssertJ Core to version 3.17.0 and use `hasDimensions`?

> hasDimensions(int expectedFirstDimension, int expectedSecondDimension): Verifies that the actual 2D array has the given dimensions.

Reply:
done

Reply:
> > Since `maze` is a jagged 2D array, wouldn't it be better to assert that all rows have the correct number of columns?
> 
> Can I update AssertJ Core to version 3.17.0 and use `hasDimensions`?
> 
> > hasDimensions(int expectedFirstDimension, int expectedSecondDimension): Verifies that the actual 2D array has the given dimensions.

Unfortunately not: the java-test-runner we use at Exercism runs offline so it has all the exercise dependencies baked in, which means we have to stick to the version of AssertJ Core that is currently used in the test runner. We can definitely look into updating AssertJ Core in the test-runner to the latest version, and then update this test case in a separate PR.